syntax = "proto3";

option java_multiple_files = true;
option java_package = "io.grpc.examples.grpcalsa";
option java_outer_classname = "GrpcAlsa";
option objc_class_prefix = "GAL";

package grpcalsa;

//Service definition
service GrpcAlsa {
  rpc PlayStream(stream AudioData) returns (PlayStatus){}
  rpc RecordStream(RecordRequest) returns (stream AudioData){}
  rpc GetStreamParameter(StreamType) returns (StreamParameter){}
}

message AudioData{
  bytes data = 1;
}

message PlayStatus{
  int32 status = 1;
}

message RecordRequest{
  int32 id = 1;
  int32 duration = 2;
}
message StreamType{
  int32 type = 1;
}
message StreamParameter{
  int32 buffersize = 1;
  int32 framerate = 2;
  int32 format = 3;
  int32 samplesize = 4;
}
